sim/create.ts

```diff
@@ -85,112 +85,116 @@ export function createWebXdc(
+          );
         }
-        const msg = `The app would now close and the user would select a chat to send this message:\nText: ${
-content.text ? `"${content.text}"` : "No Text"
-}\nFile: ${
-content.file ? `${content.file.name} - ${base64Content.length} bytes` : "No File"
-}`;
-        if (content.file) {
-            const confirmed = confirm(msg + '\n\nDownload the file in the browser instead?');
-            if (confirmed) {
-                var element = document.createElement("a");
-                element.setAttribute(
-                    "href",
-                    "data:application/octet-stream;base64," + base64Content
-                );
-                element.setAttribute("download", content.file.name);
-                document.body.appendChild(element);
-                element.click();
-                document.body.removeChild(element);
-            }
+
+        // @ts-ignore - needed because typescript imagines that blob would not exist
+        if (content.file.blob instanceof Blob) {
+          // @ts-ignore - needed because typescript imagines that blob would not exist
+          base64Content = await blob_to_base64(content.file.blob);
+          // @ts-ignore - needed because typescript imagines that base64 would not exist
+        } else if (typeof content.file.base64 === "string") {
+          // @ts-ignore - needed because typescript imagines that base64 would not exist
+          base64Content = content.file.base64;
+          // @ts-ignore - needed because typescript imagines that plainText would not exist
+        } else if (typeof content.file.plainText === "string") {
+          base64Content = await blob_to_base64(
+            // @ts-ignore - needed because typescript imagines that plainText would not exist
+            new Blob([content.file.plainText])
+          );
         } else {
-            alert(msg);
+          return Promise.reject(
+            "data is not set or wrong format, set one of `blob`, `base64` or `plainText`, see webxdc documentation for sendToChat"
+          );
         }
+      }
+      const msg = `The app would now close and the user would select a chat to send this message:\nText: ${
+        content.text ? `"${content.text}"` : "No Text"
+      }\nFile: ${
+        content.file
+          ? `${content.file.name} - ${base64Content.length} bytes`
+          : "No File"
+      }`;
+      if (content.file) {
+        const confirmed = confirm(
+          msg + "\n\nDownload the file in the browser instead?"
+        );
+        if (confirmed) {
+          var element = document.createElement("a");
+          element.setAttribute(
+            "href",
+            "data:application/octet-stream;base64," + base64Content
+          );
+          element.setAttribute("download", content.file.name);
+          document.body.appendChild(element);
+          element.click();
+          document.body.removeChild(element);
+        }
+      } else {
+        alert(msg);
+      }
```

Comment:
I'd put formatting changes and semantic changes in different commits for easier reviewing. The diff view on GitHub looks really huge, whereas it's actually all just whitespace.